src/Property.php

```diff
@@ -143,6 +143,39 @@ protected function cast($value)
         return new $castTo($value);
     }
 
+    protected function castArray(array $values)
+    {
+        if (Arr::containsType($values, DataTransferObject::class)) {
+            return $values;
+        }
+
+        $castTo = null;
+        $flatTypes = str_replace('[]', '', $this->types);
+
+        foreach ($flatTypes as $type) {
+            if (! is_subclass_of($type, DataTransferObject::class)) {
+                continue;
+            }
+
+            $castTo = $type;
+
+            break;
+        }
+
+        if (! $castTo) {
+            return $values;
+        }
+
+        return array_map(
+            function ($value) use ($castTo) {
+                if (is_array($value)) {
+                    return new $castTo($value);
+                }
+            },
```

Comment:
Ah, that's a good gotcha! To walk through my thinking: I wanted to avoid attempting to pass a non-array to the constructor of a DTO as that's the only type it it accepts. However, I agree that I don't think this is the correct approach in that it casts too wide of a net.

Perhaps a better approach would be to first validate that _all_ of the values inside of a value array were just arrays, and if one isn't throw a `DataTransferObject::invalidType` exception.